LiskHQ/lisk-sdk
# Generate standalone plugin with init:plugin command - Closes #6089

### What was the problem?

This PR resolves #6089 

### How was it solved?

- Added init:plugin command
- Added init:plugin base generator
- Added init:plugin generator
- Updated types in all files for consistency
- Added templates for standalone plugin
- Added package.json file
- Added all configuration files
- Added scripts

### How was it tested?

1. Run `yarn link` in `lisk-sdk/sdk` folder

2. Run generate standalone plugin command from commander: `./bin run init:plugin pluginName ~/path/to/plugin`

    **Note: If you run this command without the path argument e.g. ~/path/to/plugin", it will create git files in the current working commander directory and ask for overwrite. In production the user will not be running this command from the commander directory.**

3. Run `yarn link lisk-sdk` in generated plugin folder

4. Check output
